File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);

        while(!q.isEmpty()){
            TreeNode rightSide = null;
            int qLen = q.size(); 

            for(int i=0; i<qLen; i++){
                TreeNode curr = q.poll();
                if(curr!=null){
                        rightSide = curr;
                        q.add(curr.left);
                        q.add(curr.right);
                    }
            }

            if(rightSide!=null){
                res.add(rightSide.val);
            }

        }

        return res;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        if not root:
            return []

        q = deque()
        q.append(root)
        res = []

        while q:
            size = len(q)

            for i in range(0, size):
                curr = q.popleft()

                if i == size-1:
                    res.append(curr.val)

                if curr.left:
                    q.append(curr.left)
                if curr.right:
                    q.append(curr.right)

        return res


            

            